src/Equations/elastic/Kernels/DeviceAux/cuda/KernelsAux.cu

```diff
@@ -0,0 +1,335 @@
+          resultTerm[a][tid] += leftTerm[a][tid] * col;
+        }
+      }
+      __syncthreads();
+    }
+
+    if (tid < INodalDim0) {
+      for (int a = 0; a < INodalDim1; ++a) {
+        dofsFaceBoundaryNodal[tid + a * leadINodalDim] = resultTerm[a][tid] + rightTerm[a][tid];
+      }
+    }
+  }
+}
+
+
+void launchEasiBoundary(real** dofsFaceBoundaryNodalPtrs,
+                        real** easiBoundaryMapPtrs,
+                        real** easiBoundaryConstantPtrs,
+                        size_t numElements,
+                        void* deviceStream) {
+
+  dim3 block(yateto::leadDim<seissol::init::INodal>(), 1, 1);
+  dim3 grid(numElements, 1, 1);
+  auto stream = reinterpret_cast<cudaStream_t>(deviceStream);
+  kernelEasiBoundary<<<grid, block, 0, stream>>>(dofsFaceBoundaryNodalPtrs,
+                                                 easiBoundaryMapPtrs,
+                                                 easiBoundaryConstantPtrs,
+                                                 numElements);
+}
+} // seissol::kernels::local_flux::aux::details
+
+
+
+namespace seissol::kernels::time::aux {
+__global__ void kernelextractRotationMatrices(real** displacementToFaceNormalPtrs,
+                                              real** displacementToGlobalDataPtrs,
+                                              real** TPtrs,
+                                              real** TinvPtrs,
+                                              size_t numElements) {
+  const int elementId = blockIdx.x;
+  if (elementId < numElements) {
+    real* displacementToFaceNormal = displacementToFaceNormalPtrs[elementId];
+    real* displacementToGlobalData = displacementToGlobalDataPtrs[elementId];
+    auto* T = TPtrs[elementId];
+    auto* Tinv = TinvPtrs[elementId];
+
+    constexpr auto ldTinv = yateto::leadDim<seissol::init::Tinv>();
+    constexpr auto ldT = yateto::leadDim<seissol::init::T>();
+    constexpr auto ldDisplacement = yateto::leadDim<seissol::init::displacementRotationMatrix>();
+
+    const int i = threadIdx.x;
+    const int j = threadIdx.y;
+
+    displacementToFaceNormal[i + j * ldDisplacement] = Tinv[(i + 6) + (j + 6) * ldTinv];
+    displacementToGlobalData[i + j * ldDisplacement] = T[(i + 6) + (j + 6) * ldT];
+  }
+}
+
+void extractRotationMatrices(real** displacementToFaceNormalPtrs,
+                             real** displacementToGlobalDataPtrs,
+                             real** TPtrs,
+                             real** TinvPtrs,
+                             size_t numElements,
+                             void* deviceStream) {
+  dim3 block(3, 3, 1);
+  dim3 grid(numElements, 1, 1);
+  auto stream = reinterpret_cast<cudaStream_t>(deviceStream);
+  kernelextractRotationMatrices<<<grid, block, 0, stream>>>(
+    displacementToFaceNormalPtrs,
+    displacementToGlobalDataPtrs,
+    TPtrs,
+    TinvPtrs,
+    numElements);
+}
+
+__global__  void kernelInitializeTaylorSeries(real** prevCoefficientsPtrs,
```

Comment:
Maybe this should be named such that it is clear that it is the Taylor series for the grav. boundary?

Reply:
done